### PR TITLE
[Muxer] Respect deeply nested MaxItemsOne properties

### PR DESCRIPTION
**Pull request by iwahbe**
This is a follow up on https://github.com/pulumi/pulumi-terraform-bridge/pull/1015.

### CODE REVIEW

File: pf/internal/schemashim/type_schema.go [hunk 1]
```diff
@@ -56,19 +57,36 @@ func (*typeSchema) Required() bool  { return false }
 func (*typeSchema) Sensitive() bool { return false }
 
 func (s *typeSchema) Elem() interface{} {
+	asObjectType := func(typ any) (shim.Resource, bool) {
```
Comment (t0yv0) [inline pf/internal/schemashim/type_schema.go:60]: This doesn't feel right. typeSchema is in the represents a Schema that's a simple type, not block or attribute, and it's testing it for types.ObjectType on line 69, and recurring into itself. Let me check this PR out and play with this a bit.

**Comment by t0yv0** [comment]
There's an additional bug here that bic and phases marked as required when they're required.

```
            "required": [
                "automaticHttpsRewrites",
                "bic",
                "phases"
            ]
```

File: pf/tests/internal/testprovider/cmd/pulumi-resource-testbridge/schema.json [hunk 2]
```diff
@@ -48,12 +48,31 @@
...
+                    "type": "boolean"
+                },
+                "p2": {
+                    "type": "boolean"
+                }
+            },
+            "type": "object",
+            "required": [
+                "p1",
+                "p2"
```
Comment (iwahbe) [inline pf/tests/internal/testprovider/cmd/pulumi-resource-testbridge/schema.json:75]: These should be optional. I'm not 100% sure if this is an existing bug or if this change introduced a new bug. My guess is existing.

Comment (t0yv0) [reply pf/tests/internal/testprovider/cmd/pulumi-resource-testbridge/schema.json:75]: I've got a fix, one sec.

**Comment by t0yv0** [comment]
@iwahbe this passes tests on top of https://github.com/pulumi/pulumi-terraform-bridge/pull/1030 

1030 also fixes the issue with incorrectly "required" properties.

Please take a look!

I suggest we check in your PR for the awesome new test cases, but not the prod code change.

**Review by t0yv0 — APPROVED** [review]
Brilliant, let's take this.

**Pull request merged**